code/__DEFINES/dcs/flags.dm

```diff
@@ -30,6 +30,8 @@
 #define COMPONENT_DUPE_ALLOWED 1
 /// new component is deleted
 #define COMPONENT_DUPE_UNIQUE 2
+/// component uses sources
```

Comment:
Much better documentation necessary here, this is useless

Reply:
```suggestion
/**
 * Component uses source tracking to manage adding and removal logic.
 * Add a source/spawn to/the component by using AddComponentFrom(source, component_type, args...)
 * Only the first args will be respected, and you should instead handle most of your logic in the on_source_added proc.
 * Removing the last source will automatically remove the component from the parent.
 */
```